sr_hand_health_report/src/sr_hand_health_report/backlash_check.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import time
+import math
+import rospy
+from sr_hand_health_report.sr_hand_health_report_check import SrHealthReportCheck
+import numpy as np
+from urdf_parser_py.urdf import URDF
+
+
+class BacklashCheck(SrHealthReportCheck):
+
+    THRESHOLD = 0.01
+    FINGER_PWM = -250
+    WIGGLING_TIME = 2
+
+    def __init__(self, hand_side, fingers_to_test):
+        super().__init__(hand_side, fingers_to_test)
+        self._side_sign_map = {"ff": -1, "mf": -1, "rf": 1, "lf": 1, "th": 1}
+        self.joint_limits = {}
+        self._set_joint_limits()
+
+    def _set_joint_limits(self):
+        for joint in URDF.from_parameter_server().joints:
+            if joint.name.lower() in [joint_name.lower() for joint_name in self._joint_msg.name]:
```

Comment:
Your list will be created for each joint in URDF.from_parameter_server().joints, which could be avoided

Reply:
valid point, changed!